### ISSUE TITLE
Crash when running flutter run --hot, OSX with android device

### ISSUE THREAD

**Issue by tenzinjenkins** [flutter/flutter#6031]
```
W/System.err: java.io.FileNotFoundException: /sys/class/power_supply/battery/current_now: open failed: EACCES (Permission denied)
W/System.err:   at libcore.io.IoBridge.open(IoBridge.java:456)
W/System.err:   at java.io.FileInputStream.<init>(FileInputStream.java:76)
W/System.err:   at com.lock.service.chargingdetector.a.a(BatteryInfoUtil.java:344)
W/System.err:   at com.lock.service.chargingdetector.a.l(BatteryInfoUtil.java:295)
W/System.err:   at com.lock.service.chargingdetector.ChargingDetectorService$a.handleMessage(ChargingDetectorService.java:7671)
W/System.err:   at android.os.Handler.dispatchMessage(Handler.java:102)
W/System.err:   at android.os.Looper.loop(Looper.java:135)
W/System.err:   at android.os.HandlerThread.run(HandlerThread.java:61)
W/System.err: Caused by: android.system.ErrnoException: open failed: EACCES (Permission denied)
W/System.err:   at libcore.io.Posix.open(Native Method)
W/System.err:   at libcore.io.BlockGuardOs.open(BlockGuardOs.java:186)
W/System.err:   at libcore.io.IoBridge.open(IoBridge.java:442)
W/System.err:   ... 7 more
W/System.err: java.io.FileNotFoundException: /sys/class/power_supply/battery: open failed: EACCES (Permission denied)
W/System.err:   at libcore.io.IoBridge.open(IoBridge.java:456)
W/System.err:   at java.io.FileInputStream.<init>(FileInputStream.java:76)
W/System.err:   at com.lock.service.chargingdetector.a.a(BatteryInfoUtil.java:344)
W/System.err:   at com.lock.service.chargingdetector.a.l(BatteryInfoUtil.java:300)
W/System.err:   at com.lock.service.chargingdetector.ChargingDetectorService$a.handleMessage(ChargingDetectorService.java:7671)
W/System.err:   at android.os.Handler.dispatchMessage(Handler.java:102)
W/System.err:   at android.os.Looper.loop(Looper.java:135)
W/System.err:   at android.os.HandlerThread.run(HandlerThread.java:61)
W/System.err: Caused by: android.system.ErrnoException: open failed: EACCES (Permission denied)
W/System.err:   at libcore.io.Posix.open(Native Method)
W/System.err:   at libcore.io.BlockGuardOs.open(BlockGuardOs.java:186)
W/System.err:   at libcore.io.IoBridge.open(IoBridge.java:442)
W/System.err:   ... 7 more
Synced 8.6MB.
Connected to view '_flutterView/0xb4b2f080'.
Running lib/main.dart on XT1053...
E/AndroidRuntime: FATAL EXCEPTION: main
E/AndroidRuntime: Process: com.yourcompany.myapp, PID: 8029
E/AndroidRuntime: java.lang.NullPointerException: Attempt to read from field 'int io.flutter.view.AccessibilityBridge$SemanticObject.id' on a null object reference
E/AndroidRuntime:   at io.flutter.view.AccessibilityBridge.reset(AccessibilityBridge.java:315)
E/AndroidRuntime:   at io.flutter.view.FlutterView.resetAccessibilityTree(FlutterView.java:593)
E/AndroidRuntime:   at io.flutter.view.FlutterView.preRun(FlutterView.java:446)
E/AndroidRuntime:   at io.flutter.view.FlutterView.access$700(FlutterView.java:73)
E/AndroidRuntime:   at io.flutter.view.FlutterView$5.run(FlutterView.java:481)
E/AndroidRuntime:   at android.os.Handler.handleCallback(Handler.java:739)
E/AndroidRuntime:   at android.os.Handler.dispatchMessage(Handler.java:95)
E/AndroidRuntime:   at android.os.Looper.loop(Looper.java:135)
E/AndroidRuntime:   at android.app.ActivityThread.main(ActivityThread.java:5343)
E/AndroidRuntime:   at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime:   at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime:   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:905)
E/AndroidRuntime:   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:700)
W/ActivityManager:   Force finishing activity 1 com.yourcompany.myapp/org.domokit.sky.shell.SkyActivity
W/ActivityManager: Activity pause timeout for ActivityRecord{23f8e85 u0 com.yourcompany.myapp/org.domokit.sky.shell.SkyActivity t845 f}
Application finished.
^[[A^[[A^C
TenzinJenkins$ flutter run --hot
Launching loader on XT1053...
--------- beginning of main
Observatory listening on http://127.0.0.1:8108
Diagnostic server listening on http://127.0.0.1:8109
Updating assets...                                   0.1s
Syncing files to device...                               
Scanning project files...                            0.1s
Scanning package files...                            0.4s
Scanning asset files...                              0.0s
Scanning for deleted files...                        0.0s
No files to remove.
Updating files...                                    6.6s
Synced 8.6MB.
Connected to view '_flutterView/0xb4a49e40'.
Running lib/main.dart on XT1053...
E/AndroidRuntime: FATAL EXCEPTION: main
E/AndroidRuntime: Process: com.yourcompany.myapp, PID: 8213
E/AndroidRuntime: java.lang.NullPointerException: Attempt to read from field 'int io.flutter.view.AccessibilityBridge$SemanticObject.id' on a null object reference
E/AndroidRuntime:   at io.flutter.view.AccessibilityBridge.reset(AccessibilityBridge.java:315)
E/AndroidRuntime:   at io.flutter.view.FlutterView.resetAccessibilityTree(FlutterView.java:593)
E/AndroidRuntime:   at io.flutter.view.FlutterView.preRun(FlutterView.java:446)
E/AndroidRuntime:   at io.flutter.view.FlutterView.access$700(FlutterView.java:73)
E/AndroidRuntime:   at io.flutter.view.FlutterView$5.run(FlutterView.java:481)
E/AndroidRuntime:   at android.os.Handler.handleCallback(Handler.java:739)
E/AndroidRuntime:   at android.os.Handler.dispatchMessage(Handler.java:95)
E/AndroidRuntime:   at android.os.Looper.loop(Looper.java:135)
E/AndroidRuntime:   at android.app.ActivityThread.main(ActivityThread.java:5343)
E/AndroidRuntime:   at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime:   at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime:   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:905)
E/AndroidRuntime:   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:700)
W/ActivityManager:   Force finishing activity 1 com.yourcompany.myapp/org.domokit.sky.shell.SkyActivity
W/ActivityManager: Activity pause timeout for ActivityRecord{2f833d93 u0 com.yourcompany.myapp/org.domokit.sky.shell.SkyActivity t846 f}
```

```
[✓] Flutter (on Mac OS, channel alpha)
    • Flutter at /Users/camiewilson/documents/flutter
    • Framework revision 8128c56f8f (11 days ago), 2016-09-12 16:46:52
    • Engine revision 12fc138524
    • Tools Dart version 1.20.0-dev.1.0

[✓] Android toolchain - develop for Android devices (Android SDK 24.0.2)
    • Android SDK at /Users/camiewilson/Library/Android/sdk
    • Platform android-24, build-tools 24.0.2
    • Java(TM) SE Runtime Environment (build 1.8.0_101-b13)

[-] iOS toolchain - develop for iOS devices
    x XCode not installed; this is necessary for iOS development.
      Download at https://developer.apple.com/xcode/download/.
    x ideviceinstaller not available; this is used to discover connected iOS devices.
      Install via 'brew install ideviceinstaller'.
    x ios-deploy not available; this is used to deploy to connected iOS devices.
      Install via 'brew install ios-deploy'.

[✓] Atom - a lightweight development environment for Flutter
    • flutter plugin version 0.2.5
    • dartlang plugin version 0.6.39

[✓] Connected devices
    • XT1053 • T065000G26 • android-arm
```

@apwilson

**Comment by eseidelGoogle** [comment]
The top exception is some other process which probably has the wrong permissions (it's trying to read power consumption and being denied).

The later exception is Flutter's Accessibility support on Android crashing.  @jason-simmons is the right person to ask.

**Comment by chinmaygarde** [comment]
The crash seems to be that [`mFocusedObject`](https://github.com/flutter/engine/blob/master/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L315) is already null on reset.

**Comment by eseidelGoogle** [comment]
@Hixie also has some context regarding our android AX support (as well as hot reload)

**Comment by goderbauer** [comment]
Closing this as it's rather old and no steps to reproduce have been provided. If you still see this crash feel free to re-open with steps to reproduce.
